### PR TITLE
Contact Diary - Bottom Sheets (EXPOSUREAPP-4156)

### PR DESCRIPTION
**Pull request by kolyaopahle**
### Description
this PR adds the bottom sheets needed for the contact diary's day edit screen

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/person/ContactDiaryPersonBottomSheetDialogViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import de.rki.coronawarnapp.contactdiary.storage.ContactDiaryRepository
+import de.rki.coronawarnapp.ui.SingleLiveEvent
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+
+class ContactDiaryPersonBottomSheetDialogViewModel @AssistedInject constructor(
+    dispatcherProvider: DispatcherProvider,
+    private val contactDiaryRepository: ContactDiaryRepository
+) : CWAViewModel(dispatcherProvider = dispatcherProvider) {
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/person/ContactDiaryPersonBottomSheetDialogViewModel.kt:13]: How would we be able to edit an existing person?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/storage/ContactDiaryRepository.kt [hunk 2]
```diff
@@ -29,25 +29,25 @@ class ContactDiaryRepository @Inject constructor() {
...
-    suspend fun addDummyLocation() {
+    suspend fun addDummyLocation(name: String) {
         val latestElement = elements.value[0]
 
         val newElement = DefaultContactDiaryElement(
             createdAt = latestElement.createdAt,
             people = latestElement.people,
-            locations = latestElement.locations + listOf(DefaultLocation(locationName = "Somewhere"))
+            locations = latestElement.locations + listOf(DefaultLocation(locationName = name))
         )
```
Comment (chris-cwa) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/storage/ContactDiaryRepository.kt:51]: i would recommend not to use  fake data this way - create an interface and implement it with fake data first, then later, provider a for-real implementation

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/person/ContactDiaryPersonBottomSheetDialogViewModel.kt (hunk 1 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/person/ContactDiaryPersonBottomSheetDialogViewModel.kt:13]: I would've actually built a completely different bottom sheet for that as these are in the day packages and the desings do not define any edit functionality

Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/person/ContactDiaryPersonBottomSheetDialogViewModel.kt:13]: Hm, I would reuse it, especially when there are later more input fields we would have a lot of duplicate code.
The only different between both would be save vs overwrite, and a different button?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/storage/ContactDiaryRepository.kt (hunk 2 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/storage/ContactDiaryRepository.kt]: removed with usage of real room impl

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/person/ContactDiaryPersonBottomSheetDialogViewModel.kt (hunk 1 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/person/ContactDiaryPersonBottomSheetDialogViewModel.kt:14]: @kolyaopahle I can also adapt it to be reusable in my branch where I would also reuse it.

Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/person/ContactDiaryPersonBottomSheetDialogViewModel.kt:14]: Sure, would appreciate the lower workload 😅

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/location/ContactDiaryLocationBottomSheetDialogViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+
+class ContactDiaryLocationBottomSheetDialogViewModel @AssistedInject constructor(
+    dispatcherProvider: DispatcherProvider,
+    private val contactDiaryRepository: ContactDiaryRepository
+) : CWAViewModel(dispatcherProvider = dispatcherProvider) {
+
+    val shouldClose = SingleLiveEvent<Unit>()
+
+    fun saveLocation(locationName: String) = launch {
```
Comment (BMItter) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/location/ContactDiaryLocationBottomSheetDialogViewModel.kt:18]: LocationNames are only allowed with max(250chars)
could be handled in different subtask, but maybe here :-)

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/person/ContactDiaryPersonBottomSheetDialogViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    val shouldClose = SingleLiveEvent<Unit>()
+
+    fun savePerson(fullName: String) = launch {
```
Comment (BMItter) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/person/ContactDiaryPersonBottomSheetDialogViewModel.kt:18]: personNames are only allowed with max(250chars)
could be handled in different subtask, but maybe here :-)

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/location/ContactDiaryLocationBottomSheetDialogViewModel.kt (hunk 3 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/location/ContactDiaryLocationBottomSheetDialogViewModel.kt]: added validation and ui hints for field constraints

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/person/ContactDiaryPersonBottomSheetDialogViewModel.kt (hunk 4 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/person/ContactDiaryPersonBottomSheetDialogViewModel.kt]: added validation and ui hints for field constraints

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/person/ContactDiaryPersonBottomSheetDialogViewModel.kt (hunk 1 above)
Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/person/ContactDiaryPersonBottomSheetDialogViewModel.kt:17]: My idea was to either launch the bottom sheets with an argument (item ID) or without. With item ID as argument it is in "edit mode", without it is in "create new" mode.

**Pull request merged**